che-incubator/che-code
# Use Universal Developer Image as dev container

### What does this PR do?
- Inherits che-code-dev image from Universal Developer Image
- Adds few commands to build che-code image with podman

### What issues does this PR fix?

https://github.com/eclipse/che/issues/22292

### How to test this PR?
Do following:
- use the dogfooding instance to create a workspace from the current branch https://github.com/che-incubator/che-code/tree/switch-to-udi
- use devfile commands to compile and run VS Code with `yarn`
- use devfile commands to compile and then build `che-code` image with `podman`

Should have like following:

![Screenshot from 2023-06-26 17-50-43](https://github.com/che-incubator/che-code/assets/1655894/3ecc8457-7958-4ad1-b11d-028c80558320)

![Screenshot from 2023-06-26 17-49-35](https://github.com/che-incubator/che-code/assets/1655894/2c855446-9749-46a4-8108-7579d678d253)

---

### Note that:

- all the 30 commits will be squashed when merging. I haven't squashed them here because of 1) all the commits to the branch were performed using online che-code, as for me the squashing it not clear UX flow; 2) do not want to add an extra commit as it  triggers the github PR checks

- at the moment, `quay.io/che-incubator-pull-requests/che-code-dev:pr-246-dev-amd64` is used as a dev image. I plan to update it after merging of this pull request and when the corresponding che-dev image will be built by the appropriate github action.
